include/seqan3/range/view/async_buffer.hpp

```diff
@@ -0,0 +1,538 @@
+        pointer operator->() const noexcept
+        {
+            return std::addressof(cached_value);
+        }
+        //!\}
+
+        /*!\name Iterator operations
+        * \{
+        */
+        //!\brief Pre-increment.
+        async_buffer_iterator & operator++() noexcept
+        {
+            if (at_end) // TODO unlikely
+                return *this;
+
+            detail::spin_delay delay{};
+
+            assert (buffer_ptr != nullptr);
+
+            while (true)
+            {
+                switch (contrib::queue_op_status s = buffer_ptr->try_pop(cached_value); s)
+                {
+                    case contrib::queue_op_status::closed:
+                        at_end = true;
+                        return *this;
+                    case contrib::queue_op_status::success:
+                        /* cached_value was updated */
+                    #ifndef NDEBUG
+                        has_value = true;
+                    #endif
+                        return *this;
+                    default:
+                        assert(s == contrib::queue_op_status::empty);
+                        break; // loop again
+                }
+
+                delay.wait(); // pause and then try again.
+            }
+
+            // TODO [[unreachable]]
+            return *this;
+        }
+
+        //!\brief Post-increment.
+        void operator++(int) noexcept
+        {
+            ++(*this);
+        }
+        //!\}
+
+        /*!\name Comparison operators
+        * \{
+        */
+        //!\brief Compares for equality with sentinel.
+        friend constexpr bool operator==(async_buffer_iterator const & lhs,
+                                         std::ranges::default_sentinel_t const &) noexcept
+        {
+            return lhs.at_end;
+        }
+
+        //!\copydoc operator==
+        friend constexpr bool operator==(std::ranges::default_sentinel_t const &,
+                                         async_buffer_iterator const & rhs) noexcept
+        {
+            return rhs == std::ranges::default_sentinel_t{};
+        }
+
+        //!\brief Compares for inequality with sentinel.
+        friend constexpr bool operator!=(async_buffer_iterator const & lhs,
+                                         std::ranges::default_sentinel_t const &) noexcept
+        {
+            return !(lhs == std::ranges::default_sentinel_t{});
+        }
+
+        //!\copydoc operator!=
+        friend constexpr bool operator!=(std::ranges::default_sentinel_t const &,
+                                         async_buffer_iterator const & rhs) noexcept
+        {
+            return rhs != std::ranges::default_sentinel_t{};
+        }
+        //!\}
+    };
+
+public:
+    /*!\name Constructor, destructor, and assignment.
+     * \{
+     */
+    async_buffer_view()                                         = default; //!< Defaulted.
+    async_buffer_view(async_buffer_view const &)                = delete;  //!< Deleted.
```

Comment:
Usually we wanted to write a short explanation as to why it is deleted instead of `Deleted.`.
E.g., `//!< This is a move-only type.` or `//!< The queue cannot be shared.` or something like this.

Reply:
I will introduce a policy later that says that all single-passt input iterators and single-pass views are non-copyable. But I can also put it in a comment.